compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2070,7 +2073,39 @@ impl<'a> Parser<'a> {
         if has_bang {
             self.expect(&token::Not)?; // `!`
         }
-        let ident = self.parse_ident()?;
+
+        let ident = match self.parse_ident() {
+            Ok(ident) => ident,
+            Err(mut err) => {
+                match (
+                    &self.token.kind,
+                    self.look_ahead(1, |token| token.ident()),
+                    self.look_ahead(2, |token| {
+                        token.kind == TokenKind::CloseDelim(Delimiter::Parenthesis)
+                    }),
+                ) {
+                    (
+                        TokenKind::OpenDelim(Delimiter::Parenthesis),
+                        Some((Ident { span, .. }, _)),
+                        true,
+                    ) => {
+                        err.span_note(
+                            span,
+                            "try removing the parenthesis around the name for this `macro_rules!`",
+                        );
+                    }
+                    _ => {
+                        err.note(fluent::parse_maybe_missing_macro_rules_name);
```

Comment:
I understand that you've taken over the previous implementation of `MaybeMissingMacroRulesName`, but using `help` here might be a better option.

```suggestion
                        err.help(fluent::parse_maybe_missing_macro_rules_name);
```